sdcm/tester.py

```diff
@@ -2584,49 +2599,91 @@ def copy_data_between_tables(self, node, src_keyspace, src_table, dest_keyspace,
         self.log.debug('All rows have been copied from %s to %s', src_table, dest_table)
         return True
 
-    def collect_partitions_info(self, table_name, primary_key_column, save_into_file_name) -> dict | None:  # pylint: disable=too-many-locals
+    def validate_rows_per_partitions(self, skip_on_rows_number_limit: bool = True):
+        """
+        Validating partition rows-number is the same before and after running a nemesis/stress.
+        The purpose of "skip_on_rows_number_limit" is to avoid a "too heavy" scan in a too often occurrence,
+        e.g. every health check.
+        For example, if rows-per-partition is 10M and there are 50 partitions for validation,
+        It would be too much of an overhead to read 500M rows every time a nemesis is completed and health-check
+        is running. so, by default, it is limited to LIMIT_TOTAL_ROWS_NUMBER.
+        """
+        if not (self.validate_partitions and self.partitions_attributes and self.partitions_dict_before):
+            self.log.debug('Skipping validate partitions info')
+            return
+
+        if skip_on_rows_number_limit:
+            # estimated_total_rows_number represents number-of-partitions * number-of-rows-per-partition
+            # Meaning the total number of estimated rows to be read.
+            estimated_total_rows_number = len(self.partitions_dict_before) * \
+                int(list(self.partitions_dict_before.values())[0])
+            if estimated_total_rows_number > TOTAL_ROWS_NUMBER_LIMIT:
+                self.log.debug(
+                    'The estimated total number of rows (%s) exceeds allowed limit (%s). Skipping validation.',
+                    estimated_total_rows_number, TOTAL_ROWS_NUMBER_LIMIT)
+                return
+        self.log.debug('Validate partitions info')
+        partitions_dict_after = self.collect_partitions_info(partitions_attributes=self.partitions_attributes)
+        if partitions_dict_after is not None:
+            self.assertEqual(self.partitions_dict_before,
+                             partitions_dict_after,
+                             msg='Row amount in partitions is not same before and after running of nemesis')
```

Comment:
Is it possible to analyze better what is different?
IIRC, the output of the assertEqual was impossible to read.
Maybe we should dump these entire dictionaries to files + have a better way to show differences (at least first X differences).

Reply:
fixed the print in case of an assertion error.